### PR TITLE
Start ganache

### PR DESCRIPTION
**Pull request by tomclose**
This PR makes some changes to make the `yarn start` behaviour simpler and more robust.

1. If you run `yarn start` without first running `yarn start:shared-ganache`, the script will now error instead of starting an individual ganache instance.
2. If package A starts an individual ganache instance on port P, and package B then tries to join it, thinking it is a shared ganache, then the start script will error.
3. Different packages use different deploy keys, solving https://github.com/statechannels/monorepo/issues/725.
4. The puppeteer integration tests still use an individual ganache instance on circle.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -52,3 +52,5 @@ junit.xml
 
 
 ganache-deployments.json
+ganache-deployments*.json
+.ganache-deployment-cache/
```
Comment (snario) [inline .gitignore]: What folder does this get put into? Maybe if it's only in `packages/devtools` we can put this in `packages/devtools/.gitignore`.

File: packages/devtools/src/ganache/index.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {Account} from '../types';
 import {GanacheDeployer} from './deployer';
 import {GanacheNCacheDeployer} from './deployer-with-cache';
 import {GanacheServer} from './server';
+import writeJsonFile = require('write-json-file');
```
Comment (snario) [inline packages/devtools/src/ganache/index.ts]: Is import assignment necessary here?

File: packages/devtools/src/ganache/index.ts [hunk 3]
```diff
@@ -44,45 +46,88 @@ export const startOwnGanache = async (p: Partial<Params> = {}): Promise<GanacheS
...
+  const useShared = process.env.USE_GANACHE_DEPLOYMENT_CACHE === 'true';
+  const port = Number(process.env.GANACHE_PORT || 8545);
 
-  const deploymentsFile = process.env.GANACHE_DEPLOYMENTS_FILE;
-  const foundShared = await ganacheIsRunning(sharedPort);
-
-  if (!deploymentsFile) {
+  let server;
+  let deployer;
+  let type: 'shared' | 'individual';
```
Comment (snario) [inline packages/devtools/src/ganache/index.ts]: Isn't `type` a reserved keyword?

File: .gitignore (hunk 1 above)
Comment (tomclose) [reply .gitignore]: It's in the root of the monorepo.

File: packages/devtools/src/ganache/index.ts (hunk 2 above)
Comment (tomclose) [reply packages/devtools/src/ganache/index.ts]: No - I think vscode did that automatically and I didn't notice.

File: packages/devtools/src/ganache/index.ts (hunk 3 above)
Comment (tomclose) [reply packages/devtools/src/ganache/index.ts]: Yes. Weird that that works ...

**Pull request merged**